scripts/main/photo.js

```diff
@@ -1007,109 +1016,92 @@ photo.setLicense = function (photoID) {
+	if (photoIDs.length !== 1 || kind !== null) {
+		location.href = "api/Photo::getArchive?photoIDs=" + photoIDs.join() + "&kind=" + kind;
+		return;
+	}
 
-		let myPhoto;
+	// For a single photo without a specified kind, allow to pick the kind
+	// via a dialog box and re-call this method later on.
 
-		if (photo.json && photo.json.id === photoIDs[0]) {
-			myPhoto = photo.json;
-		} else {
-			myPhoto = album.getByID(photoIDs[0]);
-		}
+	const myPhoto = photo.json && photo.json.id === photoIDs[0] ? photo.json : album.getByID(photoIDs[0]);
 
-		/**
-		 * @param {string} id - the ID of the button, same semantics as "kind"
-		 * @param {string} label - the caption on the button
-		 * @returns {string} - HTML
-		 */
-		const buildButton = function (id, label) {
-			return lychee.html`
-				<a class='basicModal__button' id='${id}' title='${lychee.locale["DOWNLOAD"]}'>
-					${build.iconic("cloud-download")}${label}
+	const kind2VariantAndLocalizedLabel = {
+		FULL: ["original", lychee.locale["PHOTO_FULL"]],
+		MEDIUM2X: ["medium2x", lychee.locale["PHOTO_MEDIUM_HIDPI"]],
+		MEDIUM: ["medium", lychee.locale["PHOTO_MEDIUM"]],
+		SMALL2X: ["small2x", lychee.locale["PHOTO_SMALL_HIDPI"]],
+		SMALL: ["small", lychee.locale["PHOTO_SMALL"]],
+		THUMB2X: ["thumb2x", lychee.locale["PHOTO_THUMB_HIDPI"]],
+		THUMB: ["thumb", lychee.locale["PHOTO_THUMB"]],
+	};
+
+	/**
+	 * @param {string} kind - the kind this button is for, used to construct the ID
+	 * @returns {string} - HTML
+	 */
+	const buildButton = function (kind) {
+		return `
+				<a class='button' data-photo-kind="${kind}">
+					<svg class='iconic ionicons'><use xlink:href='#cloud-download' /></svg>
+					<span></span>
 				</a>
 			`;
-		};
+	};
 
-		let msg = lychee.html`
-			<div class='downloads'>
-		`;
+	const getPhotoArchiveDialogBody =
+		Object.entries(kind2VariantAndLocalizedLabel).reduce((html, [kind]) => html + buildButton(kind), "") + buildButton("LIVEPHOTOVIDEO");
 
-		if (myPhoto.size_variants.original.url) {
-			msg += buildButton(
-				"FULL",
-				`${lychee.locale["PHOTO_FULL"]} (${myPhoto.size_variants.original.width}x${myPhoto.size_variants.original.height},
-				${lychee.locale.printFilesizeLocalized(myPhoto.size_variants.original.filesize)})`
-			);
+	/** @param {TouchEvent|MouseEvent} ev */
+	const onClickOrTouch = function (ev) {
+		if (ev.currentTarget instanceof HTMLAnchorElement) {
+			basicModal.close();
+			photo.getArchive(photoIDs, ev.currentTarget.dataset.photoKind);
+			ev.stopPropagation();
 		}
+	};
+
+	/**
+	 * @param {ModalDialogFormElements} formElements
+	 * @param {HTMLDivElement} dialog
+	 */
+	const initGetPhotoArchiveDialog = function (formElements, dialog) {
+		Object.entries(kind2VariantAndLocalizedLabel).forEach(function ([kind, [variant, lLabel]]) {
+			/** @type {HTMLAnchorElement} */
+			const button = dialog.querySelector('a[data-photo-kind="' + kind + '"]');
+			/** @type {?SizeVariant} */
+			const sv = myPhoto.size_variants[variant];
+			if (!!sv) {
```

Comment:
Any particular reason for using `!!` here?

Reply:
No, except for my laziness to check with the specs what happens if `sv` is not `null` but `undefined`. But I guess not using `!!` is fine, too. Shall I change it?

Reply:
Fixed in https://github.com/LycheeOrg/Lychee-front/pull/322/commits/d4bdecdb2c64b19e3979cc65189fc21d8aa7e829.